control/cli.py

```diff
@@ -960,14 +961,15 @@ def listener_list(self, args):
                 for l in listeners_info.listeners:
                     adrfam = GatewayEnumUtils.get_key_from_value(pb2.AddressFamily, l.adrfam)
                     adrfam = self.format_adrfam(adrfam)
-                    listeners_list.append([l.host_name, l.trtype, adrfam, f"{l.traddr}:{l.trsvcid}"])
+                    secure = "Yes" if l.secure else "No"
+                    listeners_list.append([l.host_name, l.trtype, adrfam, f"{l.traddr}:{l.trsvcid}", secure])
```

Comment:
Just a suggestion, instead of working with plain/dumb objects (lists, tuples, dicts, etc), why not working with classes that model the main GW/CLI entities (Hosts, Subsystems, Listeners, Namespaces), and each class features specific output methods (tabular, JSON, YAML, etc):
```
class Namespace:
  def to_table(self):
    ...
  def to_json(self):
    ...
  def to_yaml(self):
    ...
```

Reply:
We should discuss it as a separate issue. It's too big to be done in the TLS PR.